dotnet-desktop-guide/net/wpf/events/attached-events-overview.md

```diff
@@ -0,0 +1,97 @@
+
+:::code language="csharp" source="./snippets/attached-events-overview/WpfControlLibraryCsharp/AquariumFilter.cs" id="AddRemoveHandlersForAttachedEvent":::
+
+:::code language="vb" source="./snippets/attached-events-overview/WpfControlLibraryVb/AquariumFilter.vb" id="AddRemoveHandlersForAttachedEvent":::
+
+The <xref:System.Windows.EventManager.RegisterRoutedEvent%2A> method that returns the attached event identifier is the same method used to register non-attached routed events. Both attached and non-attached routed events are registered to a centralized internal store. This event store implementation enables the "events as an interface" concept that's discussed in [Routed events overview](/dotnet/desktop/wpf/advanced/routed-events-overview?view=netframeworkdesktop-4.8&preserve-view=true).
+
+Unlike the CLR event "wrapper" used to back non-attached routed events, the attached event accessor methods can be implemented in classes that don't derive from <xref:System.Windows.UIElement> or <xref:System.Windows.ContentElement>. This is possible because the attached event backing code calls the <xref:System.Windows.UIElement.AddHandler%2A?displayProperty=nameWithType> and <xref:System.Windows.UIElement.RemoveHandler%2A?displayProperty=nameWithType> methods on a passed in `UIElement` instance. In contrast, the CLR wrapper for non-attached routed events calls those methods directly on the owning class, so that class must derive from `UIElement`.
+
+## Raise a WPF attached event
+
+The process for raising an attached event is essentially the same as for a non-attached routed event.
+
+Typically, your code won't need to raise any existing WPF-defined attached events since those events follow the general "service" conceptual model. In that model, service classes, such as <xref:System.Windows.Input.InputManager>, are responsible for raising WPF-defined attached events.
+
+When defining a custom attached event using the WPF model of basing attached events on [routed events](<xref:System.Windows.RoutedEvent>), use the <xref:System.Windows.UIElement.RaiseEvent%2A?displayProperty=nameWithType> method to raise an attached event on any <xref:System.Windows.UIElement> or <xref:System.Windows.ContentElement>. When raising a routed event, whether it's attached or not, you're required to designate an element in your element tree as the event source. That source is then reported as the `RaiseEvent` caller. For example, to raise the `AquariumFilter.Clean` attached routed event on `aquarium1`:
+
+:::code language="csharp" source="./snippets/attached-events-overview/csharp/MainWindow.xaml.cs" id="RaiseAttachedEvent":::
+
+:::code language="vb" source="./snippets/attached-events-overview/vb/MainWindow.xaml.vb" id="RaiseAttachedEvent":::
+
+Your code is responsible for determining which element is reported as the source in the element tree. In the preceding example, `aquarium1` is the event source.
```

Comment:
Line 91 seems to repeat what is at the end of line 85.

Reply:
@adegeo Good catch. I've removed the redundant line portion of line 91.